exercises/tom_hampson/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,189 @@
+/**
+ * Read text file and count instances of words with >= 4 letters, then save the information to a file.
+ *
+ */
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <regex>
+#include <algorithm>
+#include <map>
+#include <iomanip>
+#include <cmath>
+#include <cctype>
+
+ /** Compare number of instances of two words.
+  * @param wordA
+  * @param wordB
+  * @return True if wordA has higher count than wordB
+ */
+bool compareWordCount(const std::pair<std::string, int>& wordA, const std::pair<std::string, int>& wordB)
+{
+  return wordA.second > wordB.second;
+}
+
+/** Write the words and their counts to file.
+ *
+ * @param wordCounts
+ * @param fileName
+ * @return False if the file doesn't open.
+*/
+bool writeWordCountsToFile(std::vector<std::pair<std::string, int>>& wordCounts, const std::string& fileName)
+{
+  std::ofstream outputFile(fileName);
+
+  if (!outputFile)
+  {
+    return false;
+  }
+
+  //Something is wrong if there aren't any words.
+  if (wordCounts.size() == 0)
+  {
+    return false;
+  }
+
+  //Find length of longest word so that we can add padding to align everything nicely.
+  unsigned int longestWordLength = 0;
+  for (auto wordCount : wordCounts)
```

Comment:
I think this could be a const reference, rather than taking a copy of the value
```suggestion
  for (const auto& wordCount : wordCounts)
```

Reply:
Agreed.

Reply:
I found another instance of this too.